code/game/objects/items/weapons/RCD.dm

```diff
@@ -282,7 +288,7 @@ RCD
 
 
 /obj/item/weapon/rcd/afterattack(atom/A, mob/user, proximity)
-	if(!proximity)
+	if(!proximity && !ranged)
```

Comment:
Make this a proc and have the ARCD overide the proc and add in the ranged proc. It is much more efficent.